crud/common/utils.lua

```diff
@@ -148,6 +151,49 @@ function utils.extract_key(tuple, key_parts)
     return key
 end
 
+function utils.extract_jsonpath_keys(tuple, index_parts, condition_map)
+    if tuple == nil then
+        return nil
+    end
+
+    local key_def = nil
+    local extracted_values
+    local curr_jsonpath_ind = nil
+    local key = {}
+
+    for i, part in ipairs(index_parts) do
+        if part.path ~= nil and type(tuple[part.fieldno]) == 'table' then
+            if curr_jsonpath_ind == nil then
+                curr_jsonpath_ind = i
+            end
+
+            if key_def == nil then
+                key_def = keydef_lib.new(index_parts)
+                extracted_values = key_def:extract_key(tuple)
+            end
+
+            -- We need this check (and also condition map) for composite
+            -- jsonpath indexes. For example, we have to filtering values
+            -- by partial jsonpath index, but index is composite and fields
+            -- are jsonpaths to the same space field. In this case, we
+            -- have to extract only needed value, specified in condtion.
+            if condition_map ~= nil then
+                if condition_map[extracted_values[curr_jsonpath_ind]] then
+                    key[i] = extracted_values[curr_jsonpath_ind]
+                end
+            else
+                key[i] = extracted_values[curr_jsonpath_ind]
+            end
+
+            curr_jsonpath_ind = curr_jsonpath_ind + 1
+        else
+            key[i] = tuple[part.fieldno]
+        end
+    end
```

Comment:
I don't get why this part is needed. key_def.new(...) + key_def:extact_key/key_def:compare should be enough.